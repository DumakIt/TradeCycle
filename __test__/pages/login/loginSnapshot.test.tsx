import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { RecoilRoot } from "recoil";
import LoginPage from "../../../pages/login";
import {
  ApolloClient,
  ApolloProvider,
  HttpLink,
  InMemoryCache,
} from "@apollo/client";
import fetch from "cross-fetch";

it("login 페이지 snapshot 테스트", () => {
  const client = new ApolloClient({
    link: new HttpLink({
      uri: "http://mock.com/graphql",
      fetch,
    }),
    cache: new InMemoryCache(),
  });

  const result = render(
    <RecoilRoot>
      <ApolloProvider client={client}>
        <LoginPage />
      </ApolloProvider>
    </RecoilRoot>
  );
  expect(result.container).toMatchSnapshot();
});
