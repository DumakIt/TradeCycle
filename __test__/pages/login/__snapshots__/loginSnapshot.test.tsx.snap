// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login 페이지 snapshot 테스트 1`] = `
<div>
  <div
    class="css-i525u3"
  >
    <div>
      <div
        class="css-sabbx7"
      >
        로그인
        <span>
          Login
        </span>
      </div>
      <div
        class="css-1454229"
      />
      <form>
        <div
          class="css-12n6wyb"
        >
          <input
            class="css-yw2wmv"
            data-testid="input-email"
            name="email"
            placeholder="아이디"
            type="text"
          />
          <div
            class="css-1xo17en"
          />
        </div>
        <div
          class="css-1h42ijv"
        >
          <input
            class="css-yw2wmv"
            data-testid="input-password"
            name="password"
            placeholder="비밀번호"
            type="password"
          />
          <div
            class="css-1xo17en"
          />
        </div>
        <button
          class="css-z00cij"
          data-testid="btn-login"
        >
          로그인
        </button>
      </form>
      <div
        class="css-1njlw6g"
      >
        아직 계정이 없으신가요?
         
        <span
          data-testid="btn-signUp"
        >
          회원가입
        </span>
      </div>
    </div>
  </div>
</div>
`;
